### ISSUE TITLE
Support for .NET Core 3.1

### ISSUE THREAD

**Issue by johanhenrikssn** [aws/aws-codebuild-docker-images#277]
.NET Core [3.1](https://dotnet.microsoft.com/download/dotnet-core) is released [with LTS](https://devblogs.microsoft.com/dotnet/announcing-net-core-3-1/).

**Comment by ericbrumfield** [comment]
PR should contain what's needed to make an image on our own.  For myself I'll probably be importing it into ECR and CodeBuild for our CI/CD purposes until my team has an official one to switch to.  Hope it helps.

**Comment by su-rabbit** [comment]
When will it be available? We are waiting for this. :D

**Comment by subinataws** [comment]
@su-rabbit - This update should be releasing in the next couple of weeks. We need the updates to reach every AWS Region, before we lift the feature flag and make it visible for all users in every Region. You may consider using a custom Docker image in the meantime.

**Comment by ericbrumfield** [comment]
Interim for .Net Core 3.1.101 latest LTS that was recently released:

https://hub.docker.com/repository/docker/brummy/aws-codebuild-custom-netcore-3.1.101

**Comment by paqogomez** [comment]
Its been 20 days since @subinataws posted his message, which may qualify as a "couple of weeks".  Any update on this?

**Comment by subinataws** [comment]
@paqogomez - Thanks for checking in on the status. Devs ran into some regression and had to rollback the update that was supposed to happen yesterday. Team is working on fixing this. Apologies for the delay.

**Comment by niemyjski** [comment]
Any eta for an updated image?

**Comment by creteurlouis** [comment]
We are still waiting for this , give us some love !

**Comment by ericbrumfield** [comment]
Interim for .Net Core 3.1.102 latest LTS that was recently released:

[brummy/aws-codebuild-custom-netcore-3.1.102:3.1.2](https://hub.docker.com/repository/docker/brummy/aws-codebuild-custom-netcore-3.1.102)

**Comment by BernardLaird** [comment]
Hate to poke here, but just wondering if there has been any developments this?

**Comment by niemyjski** [comment]
This is pretty crazy.... Can everyone start filing AWS Support tickets and link to this issue so we can light a fire and at least get a generic response?

**Comment by eric-davis** [comment]
I gave up on waiting for this pretty early on and just created our own Docker image for facilitating .NET Core 3.1 builds. The Ubuntu Standard image definitions are freely available in GitHub. I just took that as a template, upgraded the .NET Core SDK install version, and removed a bunch of cruft my team doesn't use or need. Problem solved until an official AWS-supplied image is available.

**Comment by guswct** [comment]
> I gave up on waiting for this pretty early on and just created our own Docker image for facilitating .NET Core 3.1 builds. The Ubuntu Standard image definitions are freely available in GitHub. I just took that as a template, upgraded the .NET Core SDK install version, and removed a bunch of cruft my team doesn't use or need. Problem solved until an official AWS-supplied image is available.

care to share your dockerfile?

**Comment by eric-davis** [comment]
Sure. I forked the AWS build image repo into https://github.com/eric-davis/aws-codebuild-docker-images. I focused on the Ubuntu Standard 3.0 image. Updated the .NET Core SDK to version 3.1.101 and removed a bunch of stuff my team didn't need to speed up the build process and reduce the size of the resulting Docker image.

The image definition you all care about is found in the repo @ .\aws-codebuild-docker-images\ubuntu\standard\3.0. I also pushed the resulting Docker image to Docker Hub @ https://hub.docker.com/repository/docker/edavis74/aws-codebuild-docker-images-ubuntu-standard-3.0. Feel free to grab it if it's useful to you.
